src/fontra/views/fontinfo/panel-development-status-definitions.js

```diff
@@ -0,0 +1,325 @@
+      // If there is no "fontra.sourceStatusFieldDefinitions" in customData.
+      this.fontController.customData["fontra.sourceStatusFieldDefinitions"] = [];
+    }
+    const nextStatusValue = !statusFieldDefinitions
+      ? 0
+      : statusFieldDefinitions
+          .map((statusDef) => statusDef.value)
+          .sort()
+          .pop() + 1;
+
+    // fix duplicates
+    if (
+      statusDef &&
+      statusFieldDefinitions.some((statusDef) => statusDef.value === statusDef.value)
+    ) {
+      // Status definition with value already exists,
+      // changed value to next available value.
+      statusDef = {
+        ...statusDef,
+        value: nextStatusValue,
+      };
+    }
+
+    // fix isDefault
+    if (statusDef && statusFieldDefinitions.some((statusDef) => statusDef.isDefault)) {
+      // Status definition with isDefault already exists,
+      // changed value to false.
+      delete statusDef["isDefault"];
+    }
+
+    if (!statusDef) {
+      const defaultStatuses =
+        defaultStatusFieldDefinitions["fontra.sourceStatusFieldDefinitions"];
+      statusDef = defaultStatuses.find(
+        (statusDef) => statusDef.value == nextStatusValue
+      );
+    }
+
+    if (!statusDef) {
+      // No status definition provided or found.
+      // Use default.
+      statusDef = {
+        color: [1, 0, 0, 1],
+        label: `Status definition ${nextStatusValue}`,
+        value: nextStatusValue,
+      };
+    }
+
+    const undoLabel = `add status definition ${statusDef.value} '${statusDef.label}'`;
+    const root = { customData: this.fontController.customData };
+    const changes = recordChanges(root, (root) => {
+      root.customData["fontra.sourceStatusFieldDefinitions"].push(statusDef);
+    });
+    if (changes.hasChange) {
+      this.postChange(changes.change, changes.rollbackChange, undoLabel);
+      this.setupUI();
+    }
+  }
+}
+
+function getStatusFieldDefinitions(fontController) {
+  const statusFieldDefinitions =
+    fontController.customData["fontra.sourceStatusFieldDefinitions"];
+  if (statusFieldDefinitions) {
+    return statusFieldDefinitions;
+  }
+  return [];
+}
+
+addStyleSheet(`
+.fontra-ui-font-info-status-definitions-panel-status-def-box {
+  background-color: var(--ui-element-background-color);
+  border-radius: 0.5em;
+  padding: 1em;
+  cursor: pointer;
+  display: grid;
+  grid-template-rows: auto auto;
+  grid-template-columns: max-content max-content max-content auto;
+  grid-row-gap: 0.1em;
+  grid-column-gap: 1em;
+}
+
+.fontra-ui-font-info-status-definitions-panel-status-def-box-delete {
+  justify-self: end;
+  /*align-self: start;*/
+}
+
+.fontra-ui-font-info-status-definitions-panel-status-def-box-color {
+  width: 8em;
+  /*height: 1.6em; without height: best compromise for all browsers*/
+  margin: 0;
+  padding: 0;
+  outline: none;
+  border: none;
+  border-color: transparent;
+  border-radius: 0.25em;
+}
+`);
+
+class StatusDefBox extends HTMLElement {
```

Comment:
Let's call this class `StatusDefinitionBox`.

Reply:
changed it.